src/util/fs.js

```diff
@@ -454,3 +455,33 @@ export async function getFileSizeOnDisk(loc: string): Promise<number> {
 export function normalizeOS(body: string): string {
   return body.replace(/\r\n/g, '\n');
 }
+
+const cr = new Buffer('\r', 'utf8')[0];
+const lf = new Buffer('\n', 'utf8')[0];
+
+async function getEolFromFile(path: string) : Promise<string | void>  {
```

Comment:
The unfortunate bit is that it adds a read file to every write file operation.
That could unexpectedly slow down quite a few things.
Any ideas how to apply this only to yarn.lock/package.json writing?